services/workflows-service/src/workflow/workflow.controller.internal.test.ts

```diff
@@ -0,0 +1,160 @@
+import { BaseFakeRepository } from '../../../../test-utils/src/base-fake-repository';
+
+import { WorkflowControllerInternal } from './workflow.controller.internal';
+import { WorkflowService } from './workflow.service';
+import { WorkflowDefinitionModel } from './workflow-definition.model';
+import { WorkflowEventEmitterService } from './workflow-event-emitter.service';
+
+class FakeWorkflowRuntimeDataRepo extends BaseFakeRepository {
+  constructor() {
+    super(Object);
+  }
+}
+
+class FakeWorkflowDefinitionRepo extends BaseFakeRepository {
+  constructor() {
+    super(WorkflowDefinitionModel);
+  }
+}
+
+function buildWorkflowDeifintion(sequenceNum) {
+  return {
+    id: sequenceNum.toString(),
+    name: `name ${sequenceNum}`,
+    version: sequenceNum,
+    definition: {
+      initial: 'initial',
+      states: {
+        initial: {
+          on: {
+            COMPLETE: 'completed',
+          },
+        },
+        completed: {
+          type: 'final',
+        },
+      },
+    },
+    definitionType: `definitionType ${sequenceNum}`,
+    createdAt: new Date(),
+    updatedAt: new Date(),
+  };
+}
+
+describe('WorkflowControllerInternal', () => {
+  let controller;
+  let workflowRuntimeDataRepo;
+  let eventEmitterSpy;
+  const _ = Symbol('not in use');
+
+  beforeEach(() => {
+    const workflowDefinitionRepo = new FakeWorkflowDefinitionRepo();
+    workflowRuntimeDataRepo = new FakeWorkflowRuntimeDataRepo();
+
+    eventEmitterSpy = {
+      emitted: [],
+
+      emit(status, data) {
+        this.emitted.push({ status, data });
+      },
+    };
+    const service = new WorkflowService(
+      workflowDefinitionRepo as any,
+      workflowRuntimeDataRepo,
+      {} as any,
+      eventEmitterSpy as any,
+    );
+
+    controller = new WorkflowControllerInternal(service, {} as any);
+  });
+
+  describe('.listWorkflowDefinitions', () => {
+    it('returns workflows by query', async () => {
+      await controller.createWorkflowDefinition(_, buildWorkflowDeifintion(2));
+      await controller.createWorkflowDefinition(_, buildWorkflowDeifintion(3));
+      await controller.createWorkflowDefinition(_, buildWorkflowDeifintion(4));
+
+      const definitions = await controller.listWorkflowDefinitions(_, {
+        query: {
+          where: {
+            name: 'name 3',
+          },
+        },
+      });
+
+      expect(definitions).toHaveLength(1);
+      expect(definitions[0].id).toEqual('3');
```

Comment:
Can probably use a syntax that tests for a partial object instead of a line for each property

Reply:
done